File: Desafio-Front/src/types/transactionResponse.ts
export enum TransactionType
{
  DESPESA= "DESPESA",
  RECEITA = "RECEITA"
}

export enum TransactionStatus
{
  PENDENTE = "PENDENTE",
  CONCLUIDA = "CONCLUIDA",
  ATRASADA = "ATRASADA",
  CANCELADA = "CANCELADA"
}

export type TransactionResponse = {
  id: string
  description: string
  amount: string
  isActive: boolean

  peopleId: string
  categoryId: string

  transactionType: TransactionType
  status: TransactionStatus
}

export type ApiTransactionResponse = {
  success: boolean
  message: string
  data?: TransactionResponse
}

File: Desafio-WebAPI/Desafio-WebAPI/Models/Response/TransactionResponse.cs
using Desafio_WebAPI.Models.Entities;

namespace Desafio_WebAPI.Models.Response;

public class TransactionResponse
{
    public Guid Id { get; set; }
    public string Description { get; set; } = null!;
    public decimal Amount { get; set; }
    public bool? IsActive { get; set; }

    public Guid PeopleId { get; set; }
    public Guid CategoryId { get; set; }
    public Guid UserId { get; set; }
    public TransactionType TransactionType { get; set; }
    public TransactionStatus Status { get; set; }
}
